theano/gof/op.py

```diff
@@ -694,6 +694,9 @@ def __ne__(self, other):
     # Python implementation #
     #########################
 
+    def grad2(self, inputs, outputs, output_grads):
```

Comment:
As discussed a few weeks ago. I don't like the idea to have a method named grad2.

I would prefer that at the call place we have a try... except. We try with the extra args, if we get an AttributeError, then we call the old version. This would be better from a User that make Op perspective, but it have others problems.

Do you have others solution? A completly new names? I think this method don't really implement the gradient, but something close. Could this be the occasion to also please people very good in math by reconciling this other problem and not have a function ending with 2?